clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java

```diff
@@ -35,25 +36,6 @@
 
 public class LeaderAndIsrRequest extends AbstractRequest {
 
-    public static class PartitionState {
```

Comment:
If they evolve separately, it won't be code duplication anymore. Given that the duplication exists for some time and their is no definite plan of them evolving separately, why keep the duplication. Having said that if de-duping is a concern, we can leave it in, the important thing here is the `toString` override to make PartitionState log meaningful info in logs.